be/src/exec/vectorized/table_function_node.cpp

```diff
@@ -65,6 +63,7 @@ Status TableFunctionNode::init(const TPlanNode& tnode, RuntimeState* state) {
     _table_function_result_eos = true;
     _outer_column_remain_repeat_times = 0;
 
+    RETURN_IF_ERROR(ExecNode::init(tnode, state));
```

Comment:
why call parent init lastly? It is usual that call parent init first and self initial secondly. When destructing, destruct self first and destruct parent last.
In some case, there may be some members depend on parent's member which should be initialized first.
So I suggest we need to call ExecNode::init in the first part of this function.
Maybe you think it isn't a good way to call `state = INITIALIZED` in subclass, however IMO it is more important to call ExecNode::init() first. maybe you can introduce another function named set_initialized() to do this thing.

Reply:
Fine. I would separate the `state = INITIALIZED` to the tail of `init`.